src/result/error.rs

```diff
@@ -40,21 +40,21 @@ impl Error {
         Self { code, info }
     }
 
-    // Use internally to create an Error object without error info. Typically used with QueryInterface
-    // (E_NOINTERFACE) or the absense of an object to return (E_POINTER) to avoid the code gen overhead
-    // for casts that should be cheap (few instructions). Think of it as a way to create recoverable errors
-    // that don't need th overhead of debugging origination info.
-    pub fn fast_error(code: HRESULT) -> Self {
+    /// Use internally to create an Error object without error info. Typically used with [`Interface::query`]
+    /// (`E_NOINTERFACE`) or the absence of an object to return (`E_POINTER`) to avoid the
+    /// code gen overhead for casts that should be cheap (few instructions). Think of it as a way to create
+    /// recoverable errors that don't need the overhead of debugging origination info.
+    pub const fn fast_error(code: HRESULT) -> Self {
         Self { code, info: None }
     }
 
     /// The error code describing the error.
-    pub fn code(&self) -> HRESULT {
+    pub const fn code(&self) -> HRESULT {
```

Comment:
I'm not sure what value this adds as self will never be a compile-time constant.

Reply:
I had `self` as compile-time constant thanks to `pub const fn fast_error` and the example above:

```rust
impl Error {
    pub const E_POINTER: Self = Self::fast_error(HRESULT::E_POINTER);
    pub const E_NOINTERFACE: Self = Self::fast_error(HRESULT::E_NOINTERFACE);
}
```

However, it seems these are useless given that Rust starts complaining about not being able to run the destructor for `Error`, thanks to `Option<IRestrictedErrorInfo>`. It doesn't need to drop the consts as they're global/static constants, but I guess that's fallout from them being unused before?